addons/pos_self_order/static/src/mobile/pages/product_main_view/product_main_view.js

```diff
@@ -70,31 +103,27 @@ export class ProductMainView extends Component {
         if (editedLine) {
             this.state.customer_note = editedLine.customer_note;
             this.state.cartQty = editedLine.qty - 1;
-
-            if (editedLine.selected_attributes) {
-                this.state.selectedVariants = editedLine.selected_attributes;
-            }
-        } else {
-            this.state.selectedVariants = Object.fromEntries(
-                this.product.attributes.map((att) => [att.id, att.values[0].id.toString()])
-            );
         }
 
         return 0;
     }
 
+    get flattenSelectedAttribute() {
```

Comment:
Please change this into a regular function, not a getter

Reply:
There's really no reason not to do a getter here, I'm not modifying any data in this function.